lib/src/clojure_lsp/queries.clj

```diff
@@ -316,45 +316,32 @@
           (get analysis filename))))
 
 (defmethod find-references :var-usages
-  [analysis element include-declaration? _db]
+  [analysis element include-declaration? db]
   (if (= (:to element) :clj-kondo/unknown-namespace)
     [element]
-    (into []
-          (comp
-            (mapcat val)
-            (filter #(not (identical? :keywords (:bucket %))))
-            (filter #(safe-equal? (:name element) (:name %)))
-            (filter #(safe-equal? (:to element) (or (:ns %) (:to %))))
-            (filter #(or include-declaration?
-                         (and (or (not (:from-var %))
-                                  (not= (:from-var %) (:name element))
-                                  (not= (:from %) (:to element)))
-                              (not (identical? :var-definitions (:bucket %)))
-                              (not (:defmethod %)))))
-            (medley/distinct-by (juxt :filename :name :row :col)))
-          analysis)))
+    (let [var-definition {:ns (:to element)
+                          :name (:name element)
+                          :bucket :var-definitions}]
```

Comment:
Kind of dangerous, if we start to check other var-definitions fields we need to remember to update here, for example, this is missing the defined-by which is used there

Reply:
Yeah, agreed, I could only think on calling find-definition which I don't want to add right now for that until we confirm performance is not a issue there anymore, So I'm ok with that for now